Failing to build ipmi plugin on Ubuntu 21.04

**Update:** Netdata build scripts---sorry if I use the wrong term here---should be updated to look for `/usr/lib/x86_64-linux-gnu/pkgconfig/libipmimonitoring.pc/libipmimonitoring.pc` not just `/usr/lib/pkgconfig/libipmimonitoring.pc`. In the meantime you can resolve this build issue with: `sudo ln -s /usr/lib/x86_64-linux-gnu/pkgconfig/libipmimonitoring.pc/libipmimonitoring.pc /usr/lib/pkgconfig/libipmimonitoring.pc`

There is another issue which I'm going to file a new bug for where Netdata Cloud dashboard, War Room, Registry or what you call it does not update the list of charts. Which means that things like new discovered IPMI interfaces, imported or new created zpools will not show up on the Node itself when accessed through http://${ip}:19999, but not on https://app.netdata.cloud/spaces/.

---

<!--
When creating a bug report please:
- Verify first that your issue is not already reported on GitHub.
- Test if the latest release and master branch are affected too.
-->

##### Bug report summary

<!-- Provide a clear and concise description of the bug you're experiencing. -->
I tried getting the ipmi plugin to work on my machine, a desktop with a workstation motherboard (ASUS Pro WS WRX80E-Sage SE WIFI) with a recent IPMI chip. I'm running Ubuntu 21.04 at the moment <s>but I can't get it to work. It looks like the package itself had only some minor changes, but the source package has a few more entries in the changelog: http://changelogs.ubuntu.com/changelogs/pool/main/f/freeipmi/freeipmi_1.6.6-3/changelog</s>

##### OS / Environment

<!--
Provide as much information about your environment (which operating system and distribution you're using, if Netdata is running in a container, etc.)
as possible to allow us reproduce this bug faster.

To get this information, execute the following commands based on your operating system:
- uname -a; grep -Hv "^#" /etc/*release  # Linux


Place the output from the command in the code section below.  
 -->

```
$ uname -a; grep -Hv "^#" /etc/*release
Linux za31 5.11.0-25-generic #27-Ubuntu SMP Fri Jul 9 23:06:29 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=21.04
/etc/lsb-release:DISTRIB_CODENAME=hirsute
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 21.04"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="21.04 (Hirsute Hippo)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 21.04"
/etc/os-release:VERSION_ID="21.04"
/etc/os-release:HOME_URL="https://www.ubuntu.com/"
/etc/os-release:SUPPORT_URL="https://help.ubuntu.com/"
/etc/os-release:BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
/etc/os-release:PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
/etc/os-release:VERSION_CODENAME=hirsute
/etc/os-release:UBUNTU_CODENAME=hirsute
```

##### Netdata version

<!--
Provide output of `netdata -W buildinfo`.

If your Netdata Agent is older, and doesn't support buildinfo, do the following:
Provide output of `netdata -V`.
 
If Netdata is running, execute: $(ps aux | grep -E -o "[a-zA-Z/]+netdata ") -V
 -->

```
$ netdata -W buildinfo
Version: netdata v1.31.0-142-nightly
Configure options:  '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libexecdir=/usr/libexec' '--libdir=/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--enable-plugin-freeipmi' '--with-bundled-lws' 'CFLAGS=-O2' 'LDFLAGS='
Features:
    dbengine:                YES
    Native HTTPS:            YES
    Netdata Cloud:           YES 
    ACLK Next Generation:    NO
    ACLK Legacy:             YES
    TLS Host Verification:   YES
Libraries:
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    LWS:                     YES static v3.2.2
    mosquitto:               YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    NO
    EBPF:                    YES
    IPMI:                    YES
    NFACCT:                  NO
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: NO

$ $(ps aux | grep -E -o "[a-zA-Z/]+netdata ") -V
netdata v1.31.0-142-nightly
```

##### Installation method

<!--
Tell us which installation method you used (https://learn.netdata.cloud/docs/agent/packaging/installer#alternative-methods).
 -->

```
$ bash <(curl -Ss https://my-netdata.io/kickstart.sh) --reinstall --enable-plugin-freeipmi --nightly-channel
```

##### Component Name

<!--
Let us know which component is affected by the bug. Our code is structured according to its component,
so the component name is the same as the top level directory of the repository.
For example, a bug in the dashboard would be under the web component.
-->

collectors/freeipmi.plugin

##### Steps To Reproduce

<!--
Describe how you found this bug and how we can reproduce it, preferably with a minimal test-case scenario.
If you'd like to attach larger files, use gist.github.com and paste in links.
-->

1. Install Netdata.
2. Notice that while it runs on a machine with ipmi interface, ipmi charts are missing.
    - Read https://learn.netdata.cloud/docs/agent/collectors/freeipmi.plugin/
    - Searched forums and tried the following:
      ```
      $ sudo apt install build-essential freeipmi-tools libipmimonitoring-dev
      $ bash <(curl -Ss https://my-netdata.io/kickstart.sh) --reinstall --enable-plugin-freeipmi --nightly-channel
      ```
      Running `sudo ipmimonitoring` returns data.
      Installation aborts with:
      ```
      checking if apps.plugin should be enabled... yes
      checking for libipmimonitoring... no
      configure: error: ipmimonitoring required but not found. Try installing 'libipmimonitoring-dev' or 'libipmimonitoring-devel'
       FAILED

        ^
        |.-.   .-.   .-.   .-.   .-.   .  netdata                          .-.   .-
        |   '-'   '-'   '-'   '-'   '-'   sorry, it failed to build...  '-'   '-'
        +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

      Sorry! netdata failed to build...

      You may need to check these:

      1. The package uuid-dev (or libuuid-devel) has to be installed.

         If your system cannot find libuuid, although it is installed
         run me with the option:  --libs-are-really-here

      2. The package zlib1g-dev (or zlib-devel) has to be installed.

         If your system cannot find zlib, although it is installed
         run me with the option:  --libs-are-really-here

      3. The package json-c-dev (or json-c-devel) has to be installed.

         If your system cannot find json-c, although it is installed
         run me with the option:  --libs-are-really-here

      4. You need basic build tools installed, like:

         gcc make autoconf automake pkg-config

         Autoconf version 2.60 or higher is required.

      If you still cannot get it to build, ask for help at github:

         https://github.com/netdata/netdata/issues


       FAILED

       ABORTED  netdata-installer.sh exited with error
      ```
    - Compare libipmimonitoring-dev in Ubuntu 20.04 and 21.04, apparently a file has been moved, attempt to resolve this issue:
      ```
      $ sudo ln -s /usr/lib/x86_64-linux-gnu/pkgconfig/libipmimonitoring.pc/libipmimonitoring.pc /usr/lib/pkgconfig/libipmimonitoring.pc
      ```
3. Installation completes after second attempt, <s>but still no ipmi data.</s>


##### Expected behavior

<!-- Provide a clear and concise description of what you expected to happen. -->
- https://learn.netdata.cloud/docs/agent/collectors/freeipmi.plugin/ should tell what the proper command and its parameters are to reinstall and enable ipmi plugin, in my opinion. Or I'm wrong here and using the kickstart script is different than installing from source.
- <s>ipmi plugin should work after installation, of course. :-)</s>

--- a/collectors/freeipmi.plugin/README.md
+++ b/collectors/freeipmi.plugin/README.md
@@ -15,6 +15,11 @@ Netdata has a [freeipmi](https://www.gnu.org/software/freeipmi/) plugin.
 
 2.  re-install Netdata from source. The installer will detect that the required libraries are now available and will also build `freeipmi.plugin`.
 
+> In some distributions `libipmimonitoring.pc` is located in an unregistered directory.
+> In that case you should find the file and link it to the standard pkg-config directory. Usually, running
+> `sudo ln -s /usr/lib/x86_64-linux-gnu/pkgconfig/libipmimonitoring.pc/libipmimonitoring.pc /usr/lib/pkgconfig/libipmimonitoring.pc`
+> resolves the issue.
+
 Keep in mind IPMI requires root access, so the plugin is setuid to root.
 
 If you just installed the required IPMI tools, please run at least once the command `ipmimonitoring` and verify it returns sensors information. This command initialises IPMI configuration, so that the Netdata plugin will be able to work.

--- a/collectors/freeipmi.plugin/README.md
+++ b/collectors/freeipmi.plugin/README.md
@@ -15,7 +15,7 @@ Netdata has a [freeipmi](https://www.gnu.org/software/freeipmi/) plugin.
 
 2.  re-install Netdata from source. The installer will detect that the required libraries are now available and will also build `freeipmi.plugin`.
 
-> In some distributions `libipmimonitoring.pc` is located in an unregistered directory.
+> ❗ In some distributions `libipmimonitoring.pc` is located in an unregistered directory.
 > In that case you should find the file and link it to the standard pkg-config directory. Usually, running
 > `sudo ln -s /usr/lib/x86_64-linux-gnu/pkgconfig/libipmimonitoring.pc/libipmimonitoring.pc /usr/lib/pkgconfig/libipmimonitoring.pc`
 > resolves the issue.

